pkg/cmd/experimental/buildchain/buildchain.go

```diff
@@ -0,0 +1,367 @@
+				checkErr(err)
+				for _, ns := range nsList.Items {
+					namespaces = append(namespaces, ns.Name)
+				}
+			case false:
+				namespaces = append(namespaces, namespace)
+			}
+
+			// Get all build configurations
+			buildConfigList := make([]buildapi.BuildConfig, 0)
+			for _, namespace := range namespaces {
+				cfgList, err := osc.BuildConfigs(namespace).List(labels.Everything(), labels.Everything())
+				checkErr(err)
+				buildConfigList = append(buildConfigList, cfgList.Items...)
+			}
+
+			// Parse user input and validate specified image repository
+			repos := make(map[string][]string)
+			if !all && len(args) != 0 {
+				name, tag, err := parseTag(args[0])
+				checkErr(err)
+
+				// Validate the specified image repository
+				_, err = osc.ImageRepositories(namespace).Get(name)
+				checkErr(err)
+
+				repo := join(namespace, name)
+				// Set the specified repo as the only one to output dependencies for
+				repos[repo] = []string{tag}
+			} else {
+				// Get all image repositories from build configurations
+				repos = getRepos(buildConfigList)
+
+				// Make sure from now on that the --all flag is true
+				// since we are building dependency trees for every
+				// image repository available either in the current
+				// namespace or in every namespace
+				all = true
+			}
+
+			if len(repos) == 0 {
+				checkErr(fmt.Errorf("no image repository available in '%s' namespace for building its dependency tree", namespace))
+			}
+
+			output := cmdutil.GetFlagString(cmd, "output")
+			for repo, tags := range repos {
+				for _, tag := range tags {
+					root, err := findRepoDeps(repo, tag, all, buildConfigList)
+					checkErr(err)
+
+					// Check if the given image repository doesn't have any dependencies
+					if treeSize(root) < 2 {
+						glog.Infof("%s:%s has no dependencies\n", root.FullName, tag)
+						continue
+					}
+
+					switch output {
+					case "json":
+						jsonDump, err := json.MarshalIndent(root, "", "\t")
+						checkErr(err)
+						fmt.Println(string(jsonDump))
+					case "dot":
+						g := dot.NewGraph()
+						_, name, err := split(repo)
+						checkErr(err)
+						graphName := validDOT(name)
+						g.SetName(graphName)
+						// Directed graph since we illustrate dependencies
+						g.SetDir(true)
+						// Explicitly allow multiple pairs of edges between
+						// the same pair of nodes
+						g.SetStrict(false)
+						fmt.Println(dotDump(root, g, graphName))
+					case "ast":
+						fmt.Println(root)
+					default:
+						usageError(cmd, "Wrong output format specified: %s", output)
+					}
+				}
+			}
+		},
+	}
+
+	cmd.Flags().Bool("all", false, "Build dependency trees for all image repositories")
+	cmd.Flags().StringP("output", "o", "json", "Output format of dependency tree(s)")
+	return cmd
+}
+
+// getRepos iterates over a given set of build configurations
+// and extracts all the image repositories on which other
+// image repositories are based
```

Comment:
really this is "extracts all the imagerepos which trigger a buildconfig when the imagerepo is updated"